readthedocs/projects/tasks.py

```diff
@@ -571,6 +578,25 @@ def run_build(self, docker, record):
 
         if self.build_env.failed:
             self.send_notifications(self.version.pk, self.build['id'])
+            # send build failure status to git Status API
+            send_external_build_status(
+                version=self.version, build_pk=self.build['id'], status=BUILD_STATUS_FAILURE
+            )
+        elif self.build_env.successful:
+            # send build successful status to git Status API
+            send_external_build_status(
+                version=self.version, build_pk=self.build['id'], status=BUILD_STATUS_SUCCESS
+            )
+        else:
+            msg = 'Unhandled Build Status'
+            log.warning(
+                LOG_TEMPLATE,
+                {
+                    'project': self.project.slug,
+                    'version': self.version.slug,
+                    'msg': msg,
+                }
+            )
 
         build_complete.send(sender=Build, build=self.build_env.build)
```

Comment:
Instead of adding the `send_external_build_status` above, why don't we use this signal? That way, we will have everything needed once the build completes in the same place.